src/extension.ts

```diff
@@ -82,11 +82,13 @@ export class Extension implements vscode.Disposable {
 
       this.clearDecorations(editor)
 
-      editor.options.lineNumbers = vscode.TextEditorLineNumbersStyle.On
+      const lineNumberConfig = vscode.workspace.getConfiguration(extensionName).get('insertMode.lineNumbers', 'inherit');
```

Comment:
How efficient is that? I'd personally store the preference locally and update it when the setting changes.

Also, there shouldn't be semicolons (also applies further down).

Reply:
I added some utilities for automatically updating values when the configuration changes, see https://github.com/71/dance/blob/99b0c10ffa4dc4b88e3038c0fcee75169626cb99/src/extension.ts#L51-L79

You could use this to automatically update a local field when preferences change.